File: 1/1-18.ts
function multiply(a: number, b: number): number {
    if (b === 0) {
        return 0;
    } else if (b % 2 === 0) {
        // If b is even, recursively multiply a by b/2 and double the result
        return 2 * multiply(a, b / 2);
    } else {
        // If b is odd, recursively multiply a by b-1 and add a to the result
        return a + multiply(a, b - 1);
    }
}

const result = multiply(5, 7);
console.log(`Result: ${result}`);

File: 1/1-18.hs
multiply :: Int -> Int -> Int
multiply a 0 = 0
multiply a b
    | even b    = 2 * multiply a (b `div` 2)
    | otherwise = a + multiply a (b - 1)

main :: IO ()
main = do
    let result = multiply 5 7
    putStrLn $ "Result: " ++ show result